### ISSUE TITLE
PDT and IPN response received

### ISSUE THREAD

**Issue by nbdamian** [woocommerce/woocommerce#15773]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [ ] The issue still exists against the latest `master` branch of WooCommerce
- [x] This is not a usage question (Those should be directed to the [community](https://wordpress.org/support/plugin/woocommerce), unless this is a question about a premium plugin in which you should [use the helpdesk](https://woocommerce.com/my-account/tickets/) for official extensions or contact the author of 3rd party extensions)
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Make Order
2. Return to merchant website around same time as IPN
3. Both received, updates order twice.

## Expected behavior and actual behavior
When I follow those steps, I see...
Updated twice, stock reduced twice.

I was expecting...
Only once updated, only once stock decreased.

## Environment

<details>
```
### WordPress Environment ###

Home URL: https://www.fetchandmeow.com.au
Site URL: https://www.fetchandmeow.com.au
WC Version: 3.0.8
Log Directory Writable: ✔
WP Version: 4.8
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 7.0.17
PHP Post Max Size: 128 MB
PHP Time Limit: 30
PHP Max Input Vars: 4000
cURL Version: 7.45.0
OpenSSL/1.0.2h

SUHOSIN Installed: –
MySQL Version: ❌ 5.1.73 - We recommend a minimum MySQL version of 5.6. See: WordPress requirements
Max Upload Size: 128 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.0.8
WC Database Prefix: wp_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (32) ###

Gravity Forms: by rocketgenius – 2.2.3
Advanced Custom Fields: Gravity Forms Field: by @adam_pope of @stormuk – 1.0.0
AddToAny Share Buttons: by AddToAny – 1.7.11
Admin Menu Tree Page View: by Pär Thernström – 2.6.9
Advanced Custom Fields PRO: by Elliot Condon – 5.5.14
Advanced Custom Fields: Table Field: by Johann Heyne – 1.1.15
AJAX Thumbnail Rebuild: by junkcoder – 1.12
Custom Related Products for WooCommerce: by Scott Nelle – 1.3
WooCommerce Enable Free Shipping on a Per Product Basis: by Patrick Rauland & eugenf – 1.0.2
Duplicate Post: by Enrico Battocchi – 3.2
Duplicator: by Snap Creek – 1.2.8
Google Analytics for WordPress by MonsterInsights: by MonsterInsights – 6.2.0
Drop Down List Field for Gravity Forms: by Adrian Gordon – 1.7.3
Gravity Forms MailChimp Add-On: by rocketgenius – 4.2
Imsanity: by Shane Bishop – 2.3.9
Monkeyman Rewrite Analyzer: by Jan Fabry – 1.0
Bitbucket Issues: by ninetyblack – 0.2.10
Planyo online reservation system: by Xtreeme GmbH – 2.7
Post Types Order: by Nsp Code – 1.9.3.1
Radio Buttons for Taxonomies: by Kathy Darling – 1.7.7
WooCommerce Shippit: by Shippit Pty Ltd – 1.3.2
Abandoned Cart Reports For WooCommerce: by Small Fish Analytics – 2.1.10
Follow-Up Emails: by WooCommerce – 4.4.19
WooCommerce MailChimp: by Saint Systems – 2.1.0
WooCommerce Product Fees: by Caleb Burks – 1.2
WooCommerce Smart Coupons: by StoreApps – 3.3.0 – 3.3.3 is available
WooCommerce TM Extra Product Options: by themecomplete – 4.5.3
WooCommerce: by Automattic – 3.0.8 – 3.0.9 is available
WooCommerce Helper: by WooCommerce – 1.7.2 – Network enabled
Wordfence Security: by Wordfence – 6.3.11
Yoast SEO: by Team Yoast – 4.9
WP fail2ban: by Charles Lecklider – 3.5.3

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: AUD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: booking (booking)
external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #4 - /shop/
Cart: #5 - /cart/
Checkout: #6 - /checkout/
My account: #7 - /my-account/

### Theme ###

Name: Fetch & Meow
Version: 0.1
Author URL: 
Child Theme: ❌ – If you're modifying WooCommerce on a parent theme you didn't build personally
then we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: fetchandmeow/woocommerce/archive-product.php
fetchandmeow/woocommerce/auth/footer.php
fetchandmeow/woocommerce/auth/form-grant-access.php
fetchandmeow/woocommerce/auth/form-login.php
fetchandmeow/woocommerce/auth/header.php
fetchandmeow/woocommerce/cart/cart-empty.php
fetchandmeow/woocommerce/cart/cart-item-data.php
fetchandmeow/woocommerce/cart/cart-shipping.php version 2.3.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/cart/cart-totals.php
fetchandmeow/woocommerce/cart/cart.php version 2.3.8 is out of date. The core version is 3.0.3
fetchandmeow/woocommerce/cart/cross-sells.php version 1.6.4 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/cart/mini-cart.php version 2.1.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/cart/proceed-to-checkout-button.php
fetchandmeow/woocommerce/cart/shipping-calculator.php
fetchandmeow/woocommerce/checkout/cart-errors.php
fetchandmeow/woocommerce/checkout/form-billing.php version 2.1.2 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/checkout/form-checkout.php
fetchandmeow/woocommerce/checkout/form-coupon.php
fetchandmeow/woocommerce/checkout/form-login.php
fetchandmeow/woocommerce/checkout/form-pay.php version 2.4.7 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/checkout/form-shipping.php version 2.2.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/checkout/payment-method.php
fetchandmeow/woocommerce/checkout/payment.php version 2.4.7 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/checkout/review-order.php
fetchandmeow/woocommerce/checkout/thankyou.php version 2.2.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/content-product.php version 2.4.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/content-product_cat.php version 2.4.0 is out of date. The core version is 2.6.1
fetchandmeow/woocommerce/content-single-product.php version 1.6.4 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/content-widget-product.php version - is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/admin-cancelled-order.php version 2.4.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/admin-new-order.php version 2.4.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/customer-completed-order.php version 2.4.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/customer-invoice.php version 2.4.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/customer-new-account.php
fetchandmeow/woocommerce/emails/customer-note.php version 2.4.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/customer-processing-order.php version 2.4.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/customer-refunded-order.php version 2.4.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/customer-reset-password.php
fetchandmeow/woocommerce/emails/email-addresses.php version 2.4.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/emails/email-footer.php
fetchandmeow/woocommerce/emails/email-header.php
fetchandmeow/woocommerce/emails/email-order-items.php version 2.1.2 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/emails/email-styles.php
fetchandmeow/woocommerce/emails/plain/admin-cancelled-order.php version 2.3.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/plain/admin-new-order.php version 2.3.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/plain/customer-completed-order.php version 2.3.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/plain/customer-invoice.php version 2.2.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/plain/customer-new-account.php
fetchandmeow/woocommerce/emails/plain/customer-note.php version 2.3.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/plain/customer-processing-order.php version 2.3.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/plain/customer-refunded-order.php version 2.4.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/emails/plain/customer-reset-password.php
fetchandmeow/woocommerce/emails/plain/email-addresses.php
fetchandmeow/woocommerce/emails/plain/email-order-items.php version 2.1.2 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/global/breadcrumb.php
fetchandmeow/woocommerce/global/form-login.php
fetchandmeow/woocommerce/global/quantity-input.php version 2.1.0 is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/global/sidebar.php
fetchandmeow/woocommerce/global/wrapper-end.php
fetchandmeow/woocommerce/global/wrapper-start.php
fetchandmeow/woocommerce/loop/add-to-cart.php version 2.1.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/loop/loop-end.php
fetchandmeow/woocommerce/loop/loop-start.php
fetchandmeow/woocommerce/loop/no-products-found.php
fetchandmeow/woocommerce/loop/orderby.php
fetchandmeow/woocommerce/loop/pagination.php
fetchandmeow/woocommerce/loop/price.php
fetchandmeow/woocommerce/loop/rating.php version 2.0.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/loop/result-count.php version 2.0.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/loop/sale-flash.php
fetchandmeow/woocommerce/myaccount/form-add-payment-method.php version 2.1 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/myaccount/form-edit-account.php version 2.2.7 is out of date. The core version is 2.6.0
fetchandmeow/woocommerce/myaccount/form-edit-address.php version 2.1.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/myaccount/form-login.php version 2.2.6 is out of date. The core version is 2.6.0
fetchandmeow/woocommerce/myaccount/form-lost-password.php version 2.3.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/myaccount/my-account.php version 2.0.0 is out of date. The core version is 2.6.0
fetchandmeow/woocommerce/myaccount/my-address.php version 2.2.0 is out of date. The core version is 2.6.0
fetchandmeow/woocommerce/myaccount/my-downloads.php
fetchandmeow/woocommerce/myaccount/my-orders.php
fetchandmeow/woocommerce/myaccount/view-order.php version 2.2.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/notices/error.php
fetchandmeow/woocommerce/notices/notice.php
fetchandmeow/woocommerce/notices/success.php
fetchandmeow/woocommerce/order/form-tracking.php
fetchandmeow/woocommerce/order/order-again.php
fetchandmeow/woocommerce/order/order-details-customer.php version 2.4.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/order/order-details-item.php version 2.4.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/order/order-details.php version 2.4.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/order/tracking.php
fetchandmeow/woocommerce/product-searchform.php version - is out of date. The core version is 2.5.0
fetchandmeow/woocommerce/single-product/add-to-cart/external.php
fetchandmeow/woocommerce/single-product/add-to-cart/grouped.php version 2.1.7 is out of date. The core version is 3.0.7
fetchandmeow/woocommerce/single-product/add-to-cart/simple.php version 2.1.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/single-product/add-to-cart/variable.php version 2.4.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/single-product/meta.php version 1.6.4 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/single-product/price.php version 2.4.9 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/single-product/product-attributes.php version 2.1.3 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/single-product/product-image.php version 2.0.14 is out of date. The core version is 3.0.2
fetchandmeow/woocommerce/single-product/product-thumbnails.php version 2.3.0 is out of date. The core version is 3.0.2
fetchandmeow/woocommerce/single-product/rating.php
fetchandmeow/woocommerce/single-product/related.php version 1.6.4 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/single-product/review.php version 2.1.0 is out of date. The core version is 2.6.0
fetchandmeow/woocommerce/single-product/sale-flash.php
fetchandmeow/woocommerce/single-product/share.php
fetchandmeow/woocommerce/single-product/short-description.php
fetchandmeow/woocommerce/single-product/tabs/additional-information.php version 2.0.0 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/single-product/tabs/description.php
fetchandmeow/woocommerce/single-product/tabs/tabs.php
fetchandmeow/woocommerce/single-product/title.php
fetchandmeow/woocommerce/single-product/up-sells.php version 1.6.4 is out of date. The core version is 3.0.0
fetchandmeow/woocommerce/single-product-reviews.php
fetchandmeow/woocommerce/single-product.php
fetchandmeow/woocommerce/taxonomy-product_cat.php
fetchandmeow/woocommerce/taxonomy-product_tag.php

Outdated Templates: ❌Learn how to update
```
</details>

## Isolating the problem

- [ ] This bug happens with only WooCommerce plugin active
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently

**Comment by mikejolley** [comment]
This is a race condition. Disable PDT in settings to avoid it. I'm logging a separate issue to add a locking mechanism but this is not a short term fix.

**Comment by mikejolley** [comment]
See https://github.com/woocommerce/woocommerce/issues/15780

**Comment by edvinbrobeck** [comment]
We are also seeing this problem, and if disabling PDT means that we are no longer able to process/trigger refunds through WC that messes with our bookeeping. Can someone confirm that this is the case and that is what the label "Enter your PayPal API credentials to process refunds via PayPal." means.

It is to me a bit unclear if this means that we cannot trigger refunds from WC or that WC wont know if a refund was made/trigger through paypal admin

**Comment by mikejolley** [comment]
"Enter your PayPal API credentials to process refunds via PayPal."  is so you can trigger refunds in WC which go to PayPal. This is not reliant on IPN.
